azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -118,64 +95,68 @@ private void loadAllProjectFlows(Project project) {
     }
   }
 
-  public List<String> getProjectNames() {
-    return new ArrayList<String>(projectsByName.keySet());
-  }
-
   public Props getProps() {
     return props;
   }
 
   public List<Project> getUserProjects(User user) {
-    ArrayList<Project> array = new ArrayList<Project>();
-    for (Project project : projectsById.values()) {
+    ArrayList<Project> userProjects = new ArrayList<>();
+    for (Project project : getProjects()) {
       Permission perm = project.getUserPermission(user);
 
       if (perm != null
           && (perm.isPermissionSet(Type.ADMIN) || perm
               .isPermissionSet(Type.READ))) {
-        array.add(project);
+        userProjects.add(project);
       }
     }
-    return array;
+    return userProjects;
   }
 
   public List<Project> getGroupProjects(User user) {
-    List<Project> array = new ArrayList<Project>();
-    for (Project project : projectsById.values()) {
+    List<Project> groupProjects = new ArrayList<>();
+    for (Project project : getProjects()) {
       if (project.hasGroupPermission(user, Type.READ)) {
-        array.add(project);
+        groupProjects.add(project);
       }
     }
-    return array;
+    return groupProjects;
   }
 
   public List<Project> getUserProjectsByRegex(User user, String regexPattern) {
-    List<Project> array = new ArrayList<Project>();
+    List<Project> userProjects = new ArrayList<>();
     Pattern pattern;
     try {
       pattern = Pattern.compile(regexPattern, Pattern.CASE_INSENSITIVE);
     } catch (PatternSyntaxException e) {
       logger.error("Bad regex pattern " + regexPattern);
-      return array;
+      return userProjects;
     }
-
-    for (Project project : projectsById.values()) {
+    for (Project project : getProjects()) {
       Permission perm = project.getUserPermission(user);
 
       if (perm != null
           && (perm.isPermissionSet(Type.ADMIN) || perm
               .isPermissionSet(Type.READ))) {
         if (pattern.matcher(project.getName()).find()) {
-          array.add(project);
+          userProjects.add(project);
         }
       }
     }
-    return array;
+    return userProjects;
   }
 
   public List<Project> getProjects() {
-    return new ArrayList<Project>(projectsById.values());
+    List<Project> projects;
+    try {
+      projects = projectLoader.fetchAllActiveProjects();
+      for (Project proj : projects) {
```

Comment:
This is a more expensive operation. 

Is this information always needed when getProjects() is called?